app-backend/api/src/main/scala/annotation-project/Routes.scala

```diff
@@ -43,4 +157,95 @@ trait AnnotationProjectRoutes extends CommonHandlers with Authentication {
       }
     }
   }
+
+  def getAnnotationProject(projectId: UUID): Route = authenticate { user =>
+    authorizeScope(
+      ScopedAction(Domain.AnnotationProjects, Action.Read, None),
+      user
+    ) {
+      authorizeAuthResultAsync {
+        AnnotationProjectDao
+          .authorized(
+            user,
+            ObjectType.AnnotationProject,
+            projectId,
+            ActionType.View
+          )
+          .transact(xa)
+          .unsafeToFuture
+      } {
+        rejectEmptyResponse {
+          complete {
+            AnnotationProjectDao
+              .getWithRelatedAndSummaryById(projectId)
+              .transact(xa)
+              .unsafeToFuture
+          }
+        }
+      }
+    }
+  }
+
+  def updateAnnotationProject(projectId: UUID): Route = authenticate { user =>
+    authorizeScope(
+      ScopedAction(Domain.AnnotationProjects, Action.Update, None),
+      user
+    ) {
+      authorizeAuthResultAsync {
+        AnnotationProjectDao
+          .authorized(
+            user,
+            ObjectType.AnnotationProject,
+            projectId,
+            ActionType.Edit
+          )
+          .transact(xa)
+          .unsafeToFuture
+      } {
+        entity(as[AnnotationProject.WithRelated]) {
```

Comment:
I think maybe I misread what was going on in `AnnotationProjectDao`, but the point stands -- there are ids that a user can get hold of to update the specific objects, instead of our asking them for everything related to an `AnnotationProject` to update any of it.

Reply:
@jisantuc  I am suspicious about this too. But maybe it could be in its own card to be discussed and updated for all related or similar data model and api usage?

Reply:
Ah yeah I didn't mean to address it here. I just wanted to put it _somewhere_ so that next time we have a similar task we can remember that we thought/talked about it.